File: Programmers/Level1/220822/Java/KthNumberTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;

class KthNumberTest {
  @Test
  void slice() {
    KthNumber kthNumber = new KthNumber();

    assertArrayEquals(new int[]{5, 2, 6, 3},
        kthNumber.slice(new int[]{1, 5, 2, 6, 3, 7, 4}, new int[]{2, 5, 3}));
  }

  @Test
  void sort() {
    KthNumber kthNumber = new KthNumber();

    assertArrayEquals(new int[]{5},
        kthNumber.solution(
            new int[]{1, 5, 2, 6, 3, 7, 4},
            new int[][]{{2, 5, 3}})
    );
  }

  @Test
  void answer() {
    KthNumber kthNumber = new KthNumber();

    assertArrayEquals(new int[]{5, 6, 3},
        kthNumber.solution(
            new int[]{1, 5, 2, 6, 3, 7, 4},
            new int[][]{{2, 5, 3}, {4, 4, 1}, {1, 7, 3}})
    );
  }
}

File: Programmers/Level1/220823/JavaScript/KthNumber.test.js
import { solution } from './KthNumber';

test('slice', () => {
  expect(solution([1, 5, 2, 6, 3, 7, 4], [[2, 5, 3]]))
    .toStrictEqual([2, 3, 5, 6]);
});

test('solution', () => {
  expect(solution([1, 5, 2, 6, 3, 7, 4], [[2, 5, 3], [4, 4, 1], [1, 7, 3]]))
    .toStrictEqual([5, 6, 3]);
});
